### PR TITLE
add zh translation

### PR DESCRIPTION
**Pull request by codehz**

### CODE REVIEW

**Comment by codehz** [comment]
There should be no more invalid links

**Comment by kristoff-it** [comment]
Thank you for this translation, I suspect it's going to be one of the most popular ones :)

Let me know when you're done working and want me to review (tag me in a comment, or ask for a review, whichever way you prefer) and please don't esitate to ask for help if something is not clear.

Also please make sure to preview your code locally, I can be fairly confident most things are ok when using a latin alphabet but there might be some problematic situations that I'm not aware of. I'll be happy to solve any such situation for you in that case.

**Comment by codehz** [comment]
Current Progress:
```
                   | EN | IT | ZH  
-------------------+----+----+-----
  Pages            | 20 | 10 | 14  
  Paginator pages  |  0 |  0 |  0
  Non-page files   | 40 |  1 |  1
  Static files     | 11 | 11 | 11
  Processed images |  0 |  0 |  0
  Aliases          |  5 |  0 |  0
  Sitemaps         |  2 |  1 |  1
  Cleaned          |  0 |  0 |  0
```

File: content/zsf/index.zh.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ <a href="https://pex.com" rel="noopener nofollow" target="_blank"><picture>
+   <picture>
+     <source srcset="/pex-white.svg" media="(prefers-color-scheme: dark)">
+     <img src="/pex-dark.svg">
+   </picture>
+ </a>
+{{% /sponsor-logos %}}
+
+### 基础设施
+以下公司向Zig软件基金会免费提供部分服务。
```
Comment (Prince213) [inline content/zsf/index.zh.md:58]:
```suggestion
以下公司向 Zig 软件基金会免费提供部分服务。
```

File: content/_index.zh.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+---
+title: 主页
```
Comment (Prince213) [inline content/_index.zh.md:2]:
```suggestion
title: "主页"
```

File: content/_index.zh.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- 编译时调用任意函数。
+- 像操作值一样操作类型，而没有运行时成本。
+- 编译期模拟目标架构。
+
+# ⚡ 性能与安全并重
+编写快速、清晰的代码，同时处理所有错误条件。
+
+- 优雅地指导您编写错误处理逻辑。
+- 可配置的运行时检查帮助您在性能和安全保证之间取得平衡。
+- 使用可移植的向量类型表达SIMD指令。
```
Comment (Prince213) [inline content/_index.zh.md:31]:
```suggestion
- 使用可移植的向量类型表达 SIMD 指令。
```

File: content/_index.zh.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+<h1>
+	<a href="zsf/" class="button" style="display:inline;">了解更多</a>
+</h1>
+{{% /div %}}
+
+
+{{< div class="alt-background" style="padding: 20px 0;">}}
+{{% div class="container" title="赞助商" %}}
+# 企业赞助商
+以下公司为Zig软件基金会提供直接的资金支持。
```
Comment (Prince213) [inline content/_index.zh.md:105]:
```suggestion
以下公司为 Zig 软件基金会提供直接的资金支持。
```

File: content/learn/getting-started.zh.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
+---
+title: 快速入门
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:2]:
```suggestion
title: "快速入门"
```

File: content/learn/overview.zh.md [hunk 6]
```diff
@@ -0,0 +1,589 @@
...
+|        | free standing | emscripten | WASI   |
+|--------|---------------|------------|--------|
+| wasm32 | [一级支持](#一级支持)        | [三级支持](#三级支持)     | [一级支持](#一级支持) |
+| wasm64 | [四级支持](#四级支持)        | [四级支持](#四级支持)     | [四级支持](#四级支持) |
+
+
+### 支持等级
+
+#### 一级支持
+- Zig 不仅可以为这些目标生成机器代码，而且标准库的跨平台抽象也有这些目标的实现。因此，写一个不依赖 libc 的纯 Zig 应用是很实用的。
```
Comment (Prince213) [inline content/learn/overview.zh.md:547]:
```suggestion
- Zig 不仅可以为这些目标生成机器代码，而且标准库的跨平台抽象也有这些目标的实现。因此，写一个不依赖 libc 的纯 Zig 应用是可行的。
```

File: content/learn/overview.zh.md [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+toc: true
+---
+# 功能特色
+## 小巧而简洁的语言
+
+专注于调试你的应用程序，而不是调试你的编程语言知识。
+
+Zig 的完整语法可以被 [500 行的 PEG 语法的文件](https://ziglang.org/documentation/master/#Grammar)所描述。
+
+没有**隐藏的控制流**，没有隐藏的内存分配，没有预处理器，也没有宏。如果 Zig 代码看起来不像是跳开调用一个函数，那么它就不是。这意味着你可以确定下面的代码只调用 `foo()`，然后调用 `bar()`，这一点是可以保证的，而不需要知道任何东西的类型：
```
Comment (Prince213) [inline content/learn/overview.zh.md:13]:
```suggestion
没有**隐藏的控制流**，没有隐藏的内存分配，没有预处理器，也没有宏。如果 Zig 代码看起来不像是调用一个函数，那么它就不是。这意味着你可以确定下面的代码只调用 `foo()`，然后调用 `bar()`，这一点是可以保证的，而不需要知道任何东西的类型：
```

File: content/news/_index.zh.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+---
+title: 新闻
```
Comment (Prince213) [inline content/news/_index.zh.md:2]:
```suggestion
title: "新闻"
```

File: content/zsf/index.zh.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+- [Andrew Kelley](https://andrewkelley.me/) （主席）
+- [Josh Wolfe](https://github.com/thejoshwolfe/) （秘书）
+- [Mason Remaley](https://twitter.com/masonremaley/) （财务主管）
+
+## 赞助活动
+
+通过向 ZSF 捐款，您是在资助 Zig 编程语言及其生态系统的开发，从而使广大开源社区受益。Zig社区成员已经在 [LLVM](https://llvm.org/), [Wine](https://winehq.org/), [QEMU](https://qemu.org/), [musl libc](https://musl.libc.org/), [GDB](https://www.gnu.org/software/gdb/) 等项目种提交了大量 bug 的修复。
+
+ZSF是一个小组织，并且有效地利用了货币资源。我们的计划是保持这种方式，但我们确实想把我们的无偿志愿者变成有偿的维护者，以帮助合并拉取请求，使 1.0 的进展更加迅速。ZSF 作为一个非营利性的组织，其目的就是为了造福于人们。我们正在尝试让开源维护者为他们的时间获得报酬。
```
Comment (Prince213) [inline content/zsf/index.zh.md:24]:
```suggestion
ZSF 是一个小组织，并且有效地利用了货币资源。我们的计划是保持这种方式，但我们确实想把我们的无偿志愿者变成有偿的维护者，以帮助合并拉取请求，使 1.0 的进展更加迅速。ZSF 作为一个非营利性的组织，其目的就是为了造福于人们。我们正在尝试让开源维护者为他们的时间获得报酬。
```

File: content/zsf/index.zh.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+请确认您当地的法律，以确定您是否可以从您的税收中扣除捐款。
+
+|   **EIN**   | **地址** |
+|-------------|-------------|
+| 84-5105214  | Zig Software Foundation  <br> 1732 1st Ave #21385  <br> New York, NY 10128|
+
+### 其他支持的捐赠方式
+- 实物支票（见上述传统邮件地址）
+- 银行转账（包括美国以外的银行，请联系我们了解更多信息）。
+- [Benevity](https://benevity.com) （如果您的雇主想要捐款，建议使用这个方法!）
```
Comment (Prince213) [inline content/zsf/index.zh.md:39]:
```suggestion
- [Benevity](https://benevity.com) （如果你的雇主想要捐款，建议使用这个方法！）
```

File: content/zsf/index.zh.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+**如果您有任何问题或特殊需求，请毫不犹豫地与我们联系：donations@ziglang.org 。**
+
+## 企业赞助商
+
+### 货币捐款
+以下公司通过捐赠 1000 美元/月以上的方式，为Zig软件基金会提供直接的资金支持。
```
Comment (Prince213) [inline content/zsf/index.zh.md:46]:
```suggestion
以下公司通过捐赠 1000 美元/月以上的方式，为 Zig 软件基金会提供直接的资金支持。
```

**Review by Prince213 — CHANGES_REQUESTED** [review]

File: content/learn/_index.zh.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+## 在线学习资源
+- [学习 Zig](https://ziglearn.org)
+Zig 的结构化介绍，由 [Sobeston](https://github.com/sobeston) 编写。
+
+## 相关视频和博客文章
+- [Zig 1.0 之路](https://www.youtube.com/watch?v=Gv2I7qTux7g) [video]
+[Andrew Kelley](https://andrewkelley.me) 介绍Zig及其理念的视频。
+- [Zig 与 LLVM 的新关系](https://kristoff.it/blog/zig-new-relationship-llvm/)
+一篇关于构建 Zig 自托管编译器的博客文章, 也被[ lwn.net 的一篇文章所介绍](https://lwn.net/Articles/833400/).
```
Comment (Prince213) [inline content/learn/_index.zh.md:38]:
```suggestion
一篇关于构建 Zig 自托管编译器的博客文章，也被[ lwn.net 的一篇文章所介绍](https://lwn.net/Articles/833400/)。
```

File: content/learn/_index.zh.md [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+- [快速入门]({{< ref "getting-started.md" >}})
+
+## 在线学习资源
+- [学习 Zig](https://ziglearn.org)
+Zig 的结构化介绍，由 [Sobeston](https://github.com/sobeston) 编写。
+
+## 相关视频和博客文章
+- [Zig 1.0 之路](https://www.youtube.com/watch?v=Gv2I7qTux7g) [video]
+[Andrew Kelley](https://andrewkelley.me) 介绍Zig及其理念的视频。
```
Comment (Prince213) [inline content/learn/_index.zh.md:36]:
```suggestion
[Andrew Kelley](https://andrewkelley.me) 介绍 Zig 及其理念的视频。
```

File: content/learn/getting-started.zh.md [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+好消息是，从一个 Zig 版本切换到另一个版本非常容易，甚至可以在系统中同时存在多个版本。Zig 发布的版本是独立的存档，可以放在系统的任何地方。
+
+
+## 安装 Zig
+### 直接下载
+这是最直接的获取 Zig 的方法：从[下载](../../download)页面为你的平台获取压缩包，
+将其解压到一个目录，将其添加到你的 PATH 环境变量里，然后就可在任意位置调用 Zig 了。
+
+#### 在 Windows 上设置 PATH
+要在 Windows 上设置 PATH，请在 Powershell 中运行以下**任一**代码片段:
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:28]:
```suggestion
要在 Windows 上设置 PATH，请在 PowerShell 中运行以下**任一**代码片段:
```

File: content/learn/getting-started.zh.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+选择您是要在全系统级别上应用此更改 （需要在有管理员权限的情况下运行Powershell）
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:29]:
```suggestion
选择你是要在系统级别上应用此更改 （需要在有管理员权限的情况下运行 PowerShell）
```

File: content/learn/getting-started.zh.md [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+还是只针对您的用户，**并确保更改代码以指向Zig副本所在的位置**。
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:30]:
```suggestion
还是只针对你使用的用户，**并确保更改代码以指向 Zig 所在的位置**。
```

File: content/learn/getting-started.zh.md [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+注意在 `C:` 之前的 `;` 不是拼写错误。
+
+系统级安装 (**管理员** Powershell):
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:33]:
```suggestion
系统级安装 (**管理员** PowerShell):
```

File: content/learn/getting-started.zh.md [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+```
+[Environment]::SetEnvironmentVariable(
+   "Path",
+   [Environment]::GetEnvironmentVariable("Path", "Machine") + ";C:\your-path\zig-windows-x86_64-your-version",
+   "Machine"
+)
+```
+
+用户级安装 (Powershell):
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:42]:
```suggestion
用户级安装 (PowerShell):
```

File: content/learn/getting-started.zh.md [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+```
+[Environment]::SetEnvironmentVariable(
+   "Path",
+   [Environment]::GetEnvironmentVariable("Path", "User") + ";C:\your-path\zig-windows-x86_64-your-version",
+   "User"
+)
+```
+运行完毕后重启您的 Powershell 实例。
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:50]:
```suggestion
运行完毕后重启你的 PowerShell 实例。
```

File: content/learn/getting-started.zh.md [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+
+## 下一步
+**查看在 [Learn](../) 章节的其他资源**，请务必找到与你的 Zig 版本匹配的文档。
+（注意：夜间构建版本应该使用主分支上的文档），并考虑读一读 [ziglearn.org](https://ziglearn.org)。
+
+Zig 是一个年轻的项目，遗憾的是，我们还没有能力为所有的东西制作大量的文档和学习材料，因此，你应该考虑[加入其中一个 Zig 社群](https://github.com/ziglang/zig/wiki/Community)
+以在遇到问题是获取帮助，同样，也可以看看 [Zig SHOWTIME（英文）](https://zig.show)。
+
+最后，如果你喜欢 Zig 并像加速发展，[考虑捐助 Zig 软件基金会](../../zsf)
+<img src="../../heart.svg" style="vertical-align:middle; margin-right: 5px">.
```
Comment (Prince213) [inline content/learn/getting-started.zh.md:140]:
```suggestion
<img src="../../heart.svg" style="vertical-align:middle; margin-right: 5px">。
```

File: content/learn/overview.zh.md [hunk 21]
```diff
@@ -0,0 +1,589 @@
...
+$ zig build-exe --c-source hello.c --library c -target aarch64v8-linux-gnu
+$ file hello
+hello: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 2.0.0, with debug_info, not stripped
+```
+
+在某些方面，Zig 是比 C 编译器更好的 C 编译器!
+
+这个功能不仅仅是将交叉编译工具链与 Zig 捆绑在一起。例如，Zig 提供的 libc 头文件总大小为 22 MiB，未压缩。同时，仅 x86_64 上的musl libc+linux 头文件就有 8 MiB，glibc 有 3.1 MiB（glibc 缺少 linux 头文件），然而 Zig 目前附带的 libc 有 40 个。如果是天真的捆绑，那就是 444 MiB。然而，多亏了我做的这个 [process_headers](https://github.com/ziglang/zig/blob/0.4.0/libc/process_headers.zig) 工具，以及一些[老式的手工劳动](https://github.com/ziglang/zig/wiki/Updating-libc)，Zig 二进制压缩包的总容量仍然只有 30 MiB，尽管它支持所有这些目标的 libc，以及 compiler-rt、libunwind 和 libcxx，尽管它是一个 clang 兼容的 C 编译器。作为比较，来自 llvm.org 的 clang 8.0.0 本身的 Windows 二进制构建有 132 MiB 这么大。
+
+请注意，只有[一级支持](#一级支持)目标已被彻底测试。我们有计划增加[更多的 libc](https://github.com/ziglang/zig/issues/514)（包括 Windows 平台），并增加[针对所有 libc 的测试覆盖率](https://github.com/ziglang/zig/issues/2058)。
```
Comment (Prince213) [inline content/learn/overview.zh.md:414]:
```suggestion
请注意，只有[一级支持](#一级支持)目标得到了彻底测试。我们有计划增加[更多的 libc](https://github.com/ziglang/zig/issues/514)（包括 Windows 平台），并提高[对所有 libc 的测试覆盖率](https://github.com/ziglang/zig/issues/2058)。
```

File: content/learn/why_zig_rust_d_cpp.zh.md [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+
+ * 如果一个应用程序有实时性需求，那么任何使用垃圾收集或任何其他非确定性行为的库都将不予考虑。
+ * 如果一门语言让人太容易忽略错误，因此不得不验证一个库是否正确地处理和抛出错误，就很容易因此放弃这个库并重新实现它，只有自己知道自己正确地处理了所有相关的错误。Zig 的设计使程序员能做的最懒的事情就是正确处理错误，因此人们可以合理地相信一个库会正确地抛出错误。
+ * 目前，实事求是的说，C 语言是最通用、最可移植的语言。任何不具备与 C 代码互操作能力的语言都有可能被历史所抛弃。Zig 试图成为编写库的新的可移植语言，同时使导出函数直接符合 C ABI，并引入安全性和防止实现中的常见错误的语言设计。
+
+## 为现有项目的构建系统和包管理器
+
+Zig 是一门编程语言，但它也提供了一个构建系统和包管理器，即使在传统的 C/C++ 项目中也很有用。
+
+你不仅可以用 Zig 代码代替 C 或 C++ 代码，还可以用 Zig 代替 autotools、cmake、make、scons、ninja 等。而在此之上，它（将）提供一个本地依赖的包管理器。这个构建系统的目的是为了使得即使一个项目的全部代码库都是 C 或 C++ 也能适用。
```
Comment (Prince213) [inline content/learn/why_zig_rust_d_cpp.zh.md:58]:
```suggestion
你不仅可以用 Zig 代码代替 C 或 C++ 代码，还可以用 Zig 代替 autotools、cmake、make、scons、ninja 等。而在此之上，Zig（将）提供一个本地依赖的包管理器。这个构建系统的目的是为了使得即使一个项目的全部代码库都是 C 或 C++ 也能适用。
```

**Review by Prince213 — CHANGES_REQUESTED** [review]

**Review by Prince213 — APPROVED** [review]

**Comment by codehz** [comment]
@kristoff-it I think it's time to merge

**Comment by kristoff-it** [comment]
Hi @codehz, wow amazing work. I can't comment on the content itself but everything looks good from what I can tell and all links are working. 

I'll merge a fix for #99 soon. After we're done with that I'll be happy to merge everyhing.

Thank you very much again. Amazing work.

**Comment by kristoff-it** [comment]
There are 3 new variables in `i18n/en.toml`, pull from master to obtain them.  I'll list them here for convenience: 

```
[slogan-latest-release]
other = "Latest Release: &nbsp; <b>{{ . }}</b>"


[downloads-heading]
other = "Releases"

[downloads-install-from-pkg-manager]
other = "You can also <a href=\"https://github.com/ziglang/zig/wiki/Install-Zig-from-a-Package-Manager\"> install Zig from a package manager</a>."

[downloads-json]
other = "There is a <a href=\"https://ziglang.org/download/index.json\">JSON version of this page</a>."
```

Finally, please take a look at this commit, it deleted one sentence from `overvirew.md`:  https://github.com/ziglang/www.ziglang.org/commit/925f0dec9b5d1a30ae9f2e54426b5476b6495674

After this, we're done. Thank you again for your work.

**Comment by codehz** [comment]
ohhh, I messed with git-merge, you may want to use squash and merge... @kristoff-it

**Comment by codehz** [comment]
try fix it by cherry-pick

**Comment by codehz** [comment]
rebased and no conflict now

**Comment by kristoff-it** [comment]
Awesome work, thanks again. If you happen by any of the Discord communities make sure to say hi :)

**Pull request merged**